File: app/src/main/java/com/danielspeixoto/connect/module/ActivityBase.java
package com.danielspeixoto.connect.module;

import com.danielspeixoto.connect.view.activity.BaseActivity;

/**
 * Created by danielspeixoto on 1/6/17.
 */

public class ActivityBase {

    public interface View extends Base.View {

        BaseActivity getActivity();

        void goToActivity(Class clazz);
    }

    public interface Presenter extends Base.Presenter {

    }

}

File: app/src/main/java/com/danielspeixoto/connect/module/ActivityBase.kt
package com.danielspeixoto.connect.module

import com.danielspeixoto.connect.view.activity.BaseActivity

/**
 * Created by danielspeixoto on 1/6/17.
 */

class ActivityBase {

    interface View : Base.View {

        val activity: BaseActivity

        fun goToActivity(clazz: Class<*>)
    }

    interface Presenter : Base.Presenter

}